### DISCUSSION TITLE
Is there a way to include multiple PVCs in one replication object?

### DISCUSSION THREAD

**Discussion by winkrs** [backube/volsync#1115] [Q&A]
I'm using the `clusterIP` serviceType and then using the Istio Ingress Gateway to expose the replicationDestination address to the source cluster. The problem arises when I have to create one ingress gateway for every replication CR, which is not feasible considering that there can be many PVCs in a cluster. Since I do not want to use `loadBalancer` to expose the service IP, are there any other options available that I can explore to make this work? I'm considering if there is a way to group multiple PVCs in one replication object or something related to this.

**Comment by Shashankft9** [comment]
Also to add to this - the problem we are seeing is that we will need to assign a separate port to the `Gateway/VirtualService` (ingress) since its a TCP connection, so tracking this port per pvc replication between source and destination cluster could be very complex to manage.
Even if we are to use `externalIP` here, we are not sure if we will be able to afford that many IPs (considering we could have some 50 odd PVCs in a generic cluster), and then still the problem of putting the right externalIP in source cluster exists.

**Comment by tesshuflower** [comment]
There is no current way to do this - there was some investigation done a while back in terms of using CSI volume group snapshot support, which could potentially help in this situation (assuming you're trying to do multiple PVCs within the same namespace).  I realized we didn't have a tracking issue for that, so have created one:  https://github.com/backube/volsync/issues/1116

There are some details there, as well as a link to a prototype that used 1 service for 1 replicationdestination that covered multiple PVCs.

Note that CSI volume group snapshots are still at the alpha stage - we're waiting on it progressing further.  So I would consider this work stalled for the moment.
